### ISSUE TITLE
Fix S2589 FP: Rule ignores case guards

### ISSUE THREAD

**Issue by manpil** [SonarSource/sonar-dotnet#9663]
### Description

S2589 is raised on a switch expression on a tuple where the second parameter is discarded but for the first one there is a guard clause.
### Repro steps

```
 private static string Example(string input, bool y)
    {
        return (input, y) switch
        {
            ("Foo", true) => "FooTrue",
            ("Foo", false) => "FooFalse",
            (var x, true) when x.StartsWith("ab") => "AbTrue",
            (var x, _) when x.StartsWith("CD") => "CD",
            _ => "empty",
        };
    }
```

### Expected behavior

S2589 is not raised

### Actual behavior

S2589 is raised on the line containing: (var x, _) when x.StartsWith("CD") => "CD",

### Known workarounds

None/ Disable with pragma

### Related information

* C#/VB.NET Plugins version: 9.32
* Visual Studio version: N/A -> Rider 2024.2.4
* MSBuild / dotnet version: .net8.0
* SonarScanner for .NET version (if used): N/A
* Operating System: Ubuntu 22.04 LTS

Kindof relates to: https://github.com/SonarSource/sonar-dotnet/issues/8008

**Comment by sebastien-marichal** [comment]
Hello @manpil,

Thank you for reporting this issue.
I confirm this is a false positive.

I am adding this issue to our backlog to tackle it in the future.

Have a good day!

**Comment by martin-strecker-sonarsource** [comment]
Tracked by https://sonarsource.atlassian.net/browse/NET-1208 now

**Comment by mary-georgiou-sonarsource** [comment]
Fixed in sonar-dotnet 10.9
